desc/integrals/bounce_utils.py

```diff
@@ -27,36 +27,34 @@
 )
 
 
-def get_pitch_inv(min_B, max_B, num, relative_shift=1e-6):
-    """Return 1/λ values for quadrature between ``min_B`` and ``max_B``.
+def get_pitch_inv_quad(min_B, max_B, num_pitch):
+    """Return 1/λ values and weights for quadrature between ``min_B`` and ``max_B``.
 
     Parameters
     ----------
     min_B : jnp.ndarray
         Minimum |B| value.
     max_B : jnp.ndarray
         Maximum |B| value.
-    num : int
-        Number of values, not including endpoints.
-    relative_shift : float
-        Relative amount to shift maxima down and minima up to avoid floating point
-        errors in downstream routines.
+    num_pitch : int
+        Number of values.
 
     Returns
     -------
-    pitch_inv : jnp.ndarray
-        Shape (*min_B.shape, num + 2).
-        1/λ values.
+    pitch_inv, weight : (jnp.ndarray, jnp.ndarray)
+        Shape (*min_B.shape, num_pitch).
+        1/λ values and weights.
 
     """
-    # Floating point error impedes consistent detection of bounce points riding
-    # extrema. Shift values slightly to resolve this issue.
-    min_B = (1 + relative_shift) * min_B
-    max_B = (1 - relative_shift) * max_B
+    errorif(
+        num_pitch > 1e5,
+        msg=f"{num_pitch} > 1e5 not possible due to floating point errors.",
```

Comment:
can you explain this a bit? Where do the errors pop up?

Reply:
This replaces the relative shift logic above. if a given pitch_inv value is too close to global min and max B, then floating point errors will make it impossible to detect whether a given pitch horizontal line intersects B.

Reply:
i’ll add back the comment